client/state/data-layer/wpcom/read/tags/index.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Internal dependencies
+ */
+import {
+	READER_FETCH_TAGS_REQUEST,
+} from 'state/action-types';
+import {
+	receiveTags,
+} from 'state/reader/tags/items/actions';
+
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+
+export function requestTags( store, action, next ) {
+	const path = action.payload && action.payload.tag
```

Comment:
and these are wildly different actions. it's going to make using this really weird.